build_config/lib/src/build_config.dart

```diff
@@ -12,6 +12,37 @@ import 'package:path/path.dart' as p;
 import 'parse.dart';
 import 'pubspec.dart';
 
+/// A glob filter on files inputs or sources.
+///
+/// Takes a list of strings in glob format for [include] and [exclude]. Matches
+/// the `glob()` function in skylark.
+class InputSet {
+  /// The globs to include in the set.
+  ///
+  /// May be null or empty which means every possible path (like `'**'`).
+  final List<String> include;
+
+  /// The globs as a subset of [include] to remove from the set.
+  ///
+  /// May be null or empty which means every path in [include].
+  final List<String> exclude;
+  const InputSet({this.include, this.exclude});
+
+  @override
+  String toString() {
+    final result = new StringBuffer()..write('paths matching $include');
```

Comment:
I don't think you want the `write` call here? (looks like its handled better right below)

Reply:
yup, broken refactor. Fixed.